### PR TITLE
06-add-user-role-user_use_cases-management

### PR DESCRIPTION
**Pull request by andrey-kiyas**
Added:
- User
- Role
- User Use Cases
- Management

### CODE REVIEW

File: src/main/java/com/onlinebookstore/dto/user/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class UserLoginRequestDto {
+    @NotNull
+    @Size(min = 4, max = 50)
+    @Email
+    private String email;
```
Comment (aaananas) [inline src/main/java/com/onlinebookstore/dto/user/UserLoginRequestDto.java]:
```suggestion
    @Email
    private String email;
```
check the length and if null in email validator

File: src/main/java/com/onlinebookstore/dto/user/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@FieldMatch(
+        field = "password",
+        fieldMatch = "repeatPassword",
+        message = "Password and repeat password shouldn't be empty and should be equal"
+)
+public class UserRegistrationRequestDto {
+    @NotNull
+    @Size(min = 4, max = 50)
+    @Email
+    private String email;
```
Comment (aaananas) [inline src/main/java/com/onlinebookstore/dto/user/UserRegistrationRequestDto.java]:
```suggestion
    @Email
    private String email;
```

**Review by aaananas — CHANGES_REQUESTED** [review]
requested changes

File: src/main/java/com/onlinebookstore/dto/user/UserRegistrationRequestDto.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/dto/user/UserRegistrationRequestDto.java]: Yes, the condition is redundant, even in the absence of validator. Got it. Thank you!

File: src/main/java/com/onlinebookstore/dto/user/UserLoginRequestDto.java (hunk 1 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/dto/user/UserLoginRequestDto.java]: Thank you!

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 3]
```diff
@@ -23,36 +24,41 @@
 @Tag(name = "Book management", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
-@RequestMapping(value = "/api/books")
+@RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
 
     @Operation(summary = "Find all books", description = "Find all books")
+    @PreAuthorize("isAuthenticated()")
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/controller/BookController.java:32]: you already have this config 
image

**Review by Rommelua — APPROVED** [review]

**Pull request merged**